src/Kafka/ManageKafkaPermissions/components/SelectAccount.tsx

```diff
@@ -0,0 +1,187 @@
+            )}
+          >
+            {t("manage_permissions_dialog.all_accounts_title")}
+          </SelectOption>
+        </SelectGroup>,
+        <Divider key="all_accounts_divider" />,
+        <SelectGroup
+          label={t(
+            "manage_permissions_dialog.all_accounts_service_account_group"
+          )}
+          key="service_accounts_group"
+        >
+          {initialOptions
+            .filter(
+              (principal) =>
+                principal.principalType === PrincipalType.ServiceAccount
+            )
+            .sort((a, b) =>
+              a.displayName && b.displayName
+                ? a.displayName.localeCompare(b.displayName)
+                : -1
+            )
+            .map((principal, index) => (
+              <SelectOption
+                key={index}
+                value={principal.id}
+                description={principal.displayName}
+              >
+                {principal.id}
+              </SelectOption>
+            ))}
+        </SelectGroup>,
+        <Divider key="user_account_divider" />,
+        <SelectGroup
+          label={t("manage_permissions_dialog.all_accounts_user_account_group")}
+          key="user_accounts_group"
+        >
+          {initialOptions
+            .filter(
+              (principal) =>
+                principal.principalType === PrincipalType.UserAccount
+            )
+            .map((principal, index) => (
+              <SelectOption
+                key={index}
+                value={principal}
+                description={principal.displayName}
+              >
+                {principal.id}
+              </SelectOption>
+            ))}
+        </SelectGroup>,
+      ]
+    : [];
+
+  const customFilter = (
+    _: React.ChangeEvent<HTMLInputElement> | null,
+    value: string
+  ) => {
+    if (!value) {
+      return options;
+    }
+
+    const input = new RegExp(value, "i");
+    return options
+      .filter((accounts) => Array.isArray(accounts.props.children))
+      .map((account) =>
+        account.props.children.filter(
+          (allAccounts: { props: { value: string; description: string } }) =>
+            input.test(allAccounts.props.value) ||
+            input.test(allAccounts.props.description)
+        )
+      );
+  };
+
+  return (
+    <FormGroupWithPopover
+      labelHead={t("manage_permissions_dialog.account_id_title")}
+      fieldId="kafka-instance-name"
+      fieldLabel={t("manage_permissions_dialog.account_id_title")}
+      labelBody={t("manage_permissions_dialog.account_id_help")}
+      buttonAriaLabel={t("manage_permissions_dialog.account_id_aria")}
+      isRequired={true}
+      helperTextInvalid={id.errorMessage}
+      validated={id.validated || ValidatedOptions.default}
+    >
+      <Select
+        data-testid="acls-select-account"
+        variant={SelectVariant.typeahead}
+        className="kafka-ui--select--limit-height"
```

Comment:
A CSS class without a companion rule? Please remove this if we don't need it, and if we do there is a [naming convention that needs to be followed](https://redhat-developer.github.io/app-services-ui-components/iframe.html?id=contributing--page&args=&viewMode=story#classnames-and-id-naming-conventions) when naming classes